cmd/algoh/blockstats_test.go

```diff
@@ -57,17 +57,36 @@ func TestConsecutiveBlocks(t *testing.T) {
 }
 
 func TestAgreementTime(t *testing.T) {
-	sender := MockEventSender{}
-	bs := blockstats{log: &sender}
+	sleepTime := 50 * time.Millisecond
+	testAttempts := 0
+	const maxTestAttempts = 10
+	for {
+		sleepTimeHighWatermark := time.Duration(int64(sleepTime) * 105 / 100)
 
-	bs.onBlock(v1.Block{Round: 300})
-	time.Sleep(500 * time.Millisecond)
-	bs.onBlock(v1.Block{Round: 301})
+		sender := MockEventSender{}
+		bs := blockstats{log: &sender}
+
+		start := time.Now()
+		bs.onBlock(v1.Block{Round: 300})
+		time.Sleep(sleepTime)
+		bs.onBlock(v1.Block{Round: 301})
+		end := time.Now()
+
+		require.Equal(t, 1, len(sender.events))
+		details := sender.events[0].details.(telemetryspec.BlockStatsEventDetails)
+
+		// Make sure the duration is close to 500ms
```

Comment:
not actually 500ms but `sleepTime` which is now 50ms ?

Reply:
exactly. The goal of this test is to see that onBlock measure the time difference.
There is no reason to wait 500ms for that if the machine is reasonably fast. On the other hand, on slow machine, this is unlikely to be enough.
So, I changed the timeout to be +5% over the sleep time, and tested the worst case scenario by wrapping the two called between start/end timers.
That allows faster machines to test that quickly, and would take longer on slower machines.

Reply:
yeah, I was just commenting that cleaning up the comment to be more accurate would be nice

Reply:
oh, I completely missing that.. I'll update the comment to reflect what's being done.

thanks!